cunumeric/module.py

```diff
@@ -2442,6 +2503,156 @@ def diag_indices_from(arr: ndarray) -> tuple[ndarray, ...]:
     return diag_indices(arr.shape[0], arr.ndim)
 
 
+def tril_indices(
+    n: int, k: int = 0, m: Optional[int] = None
+) -> tuple[ndarray, ...]:
+    """
+    Return the indices for the lower-triangle of an (n, m) array.
+
+    Parameters
+    ----------
+    n : int
+        The row dimension of the arrays for which the returned
+        indices will be valid.
+    k : int, optional
+        Diagonal offset (see `tril` for details).
+    m : int, optional
+        The column dimension of the arrays for which the returned
+        arrays will be valid.
+        By default `m` is taken equal to `n`.
+
+    Returns
+    -------
+    inds : tuple of arrays
+        The indices for the triangle. The returned tuple contains two arrays,
+        each with the indices along one dimension of the array.
+
+    See also
+    --------
+    numpy.tril_indices
+
+    Notes
+    -----
+
+    Availability
+    ------------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    tri_ = tri(n, m, k=k, dtype=bool)
+    return nonzero(tri_)
+
+
+@add_boilerplate("arr")
+def tril_indices_from(arr: ndarray, k: int = 0) -> tuple[ndarray, ...]:
+    """
+    Return the indices for the lower-triangle of arr.
+
+    See `tril_indices` for full details.
+
+    Parameters
+    ----------
+    arr : array_like
+        The indices will be valid for square arrays whose dimensions are
```

Comment:
AFAICT the implementation works for non-square arrays as well, so I don't think we need to require "square" here, similar for `triu_indices_from`.